### PR TITLE
*: fix unit tests on the windows platform

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

My linux crash recently, so maybe I have to develop on windows for a while.
(Emacs + Chrome + msys2 should cover 90% of my daily usage, good news)

First of all, I'd like to fix the unit tests on windows.

### What is changed and how it works?

TiDB is written in Go, which is a great cross-platform language.
I find it runs smoothly on windows, without much modification.

Some behaviors I found that Go differs between windows and linux:

1. time.Now() is imprecise on windows, time.Now().UnixNano() doesn't work well.
2. os.Remove()  returns an error if a file is open and not yet closed.
3. The unix socket address can't be "host:port"
4. Some command like `sysctl` is missing, of course 
5. The handle of \n as a newline is different on windows

### Check List 

Tests 

 - Unit test
 - No code

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @lysu @jackysp

File: util/logutil/log_test.go [hunk 1]
```diff
@@ -168,6 +169,13 @@ func (s *testLogSuite) TestLoggerKeepOrder(c *C) {
 }
 
 func (s *testLogSuite) TestSlowQueryZapLogger(c *C) {
+	if runtime.GOOS == "windows" {
+		// Skip this test on windows for two reasons:
+		// 1. The pattern match fails somehow. It seems windows treat \n as slash and charactor n.
```
Comment (lysu) [inline util/logutil/log_test.go]:
```suggestion
		// 1. The pattern match fails somehow. It seems windows treat \n as slash and character n.
```

File: util/logutil/log_test.go [hunk 2]
```diff
@@ -197,6 +205,13 @@ func (s *testLogSuite) TestSlowQueryZapLogger(c *C) {
 }
 
 func (s *testLogSuite) TestZapLoggerWithKeys(c *C) {
+	if runtime.GOOS == "windows" {
+		// Skip this test on windows for two reason:
+		// 1. The pattern match fails somehow. It seems windows treat \n as slash and charactor n.
```
Comment (lysu) [inline util/logutil/log_test.go]:
```suggestion
		// 1. The pattern match fails somehow. It seems windows treat \n as slash and character n.
```
fix CI lint

**Comment by tiancaiamao** [comment]
/rebuild

```
[2020-02-18T03:59:56.852Z] {"level":"warn","ts":"2020-02-18T11:59:45.436+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target"
:"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":4,"error":"rpc error: code = Canceled desc = grpc: the client connection
is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.463+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target"
:"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":5,"error":"rpc error: code = Canceled desc = grpc: the client connection
is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.487+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":6,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.513+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":7,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.536+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":8,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.560+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":9,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.584+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":10,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.607+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":11,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] {"level":"warn","ts":"2020-02-18T11:59:45.631+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-031e6428-9425-46fb-a435-b8ec0ae1060d/localhost:91133613029534266410","attempt":12,"error":"rpc error: code = Canceled desc = grpc: the client connection is closing"}
[2020-02-18T03:59:56.853Z] --- FAIL: TestSyncerSimple (28.77s)
[2020-02-18T03:59:56.853Z]     syncer_test.go:120: get udpate version failed
[2020-02-18T03:59:56.853Z] FAIL
[2020-02-18T03:59:56.853Z] coverage: 53.1% of statements
[2020-02-18T03:59:56.853Z] FAIL github.com/pingcap/tidb/ddl/util        28.809s
```

https://github.com/pingcap/tidb/issues/14777

**Comment by tiancaiamao** [comment]
PTAL @lysu @jackysp

**Pull request merged**